src/main/java/com/djrapitops/extension/DKCoinsExtension.java

```
@@ -0,0 +1,97 @@
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.djrapitops.extension;
+
+import ch.dkrieger.coinsystem.core.CoinSystem;
+import ch.dkrieger.coinsystem.core.player.CoinPlayer;
+import com.djrapitops.plan.extension.CallEvents;
+import com.djrapitops.plan.extension.DataExtension;
+import com.djrapitops.plan.extension.annotation.*;
+import com.djrapitops.plan.extension.icon.Color;
+import com.djrapitops.plan.extension.icon.Family;
+
+import java.util.Optional;
+import java.util.UUID;
+
+/**
+ * DataExtension for the DKCoins
+ *
+ * @author Vankka
+ */
+@PluginInfo(name = "DKCoins", iconName = "coins", iconFamily = Family.SOLID, color = Color.YELLOW)
+public class DKCoinsExtension implements DataExtension {
+
+    public DKCoinsExtension() {
+    }
+
+    @Override
+    public CallEvents[] callExtensionMethodsOn() {
+        return new CallEvents[]{ CallEvents.PLAYER_JOIN };
+    }
+
+    private CoinSystem getCoinSystem() {
+        return CoinSystem.getInstance();
+    }
+
+    private Optional<CoinPlayer> getCoinPlayer(UUID playerUUID) {
+        return Optional.ofNullable(getCoinSystem().getPlayerManager().getPlayer(playerUUID));
+    }
+
+    @BooleanProvider(
+            text = "",
+            hidden = true,
+            conditionName = "playerLoaded"
+    )
+    public boolean playerLoaded(UUID playerUUID) {
+        return getCoinPlayer(playerUUID).isPresent();
+    }
+
+    @Conditional("playerLoaded")
```

You can throw a `NotReadyException` if the player is not present (to avoid the extra conditional)